### PR TITLE
John updates

### PR DESCRIPTION
**Pull request by J-hon**

### CODE REVIEW

File: app/Jobs/BookReturnReminderJob.php [hunk 1]
```diff
@@ -34,16 +36,47 @@ public function __construct()
...
+        foreach ($ledges as $ledge)
+        {
+            $this->sendReminderMail($ledge);
         }
     }
 
-    private function differenceBetweenTodayAndDate($date)
+    private function sendReminderMail(Ledge $ledge)
+    {
+        if ($ledge->status === LedgeStatus::WaitingPickup && $ledge->return_date !== NULL && $this->differenceInDaysBetweenTodayAndDate($ledge->return_date) === 1)
```
Comment (Zelig880) [inline app/Jobs/BookReturnReminderJob.php:47]: There are two things to mention here:
- [ ] we should use the pickup_date not the return date 
- [ ] We should send a different email (as this is a reminder to pick the book up, not the one to return it. I am ok if you just make a copy of the return for now), but they have to be different as they will have quite a different language

File: app/Console/Kernel.php [hunk 2]
```diff
@@ -26,9 +26,9 @@ class Kernel extends ConsoleKernel
      */
     protected function schedule(Schedule $schedule)
     {
-        $schedule->job(new PickupBookJob())->dailyAt('12:00');
+        $schedule->job(new PickupBookJob())->hourly();
 
-        $schedule->job(new BookReturnReminderJob())->dailyAt('12:00');
+        $schedule->job(new BookReturnReminderJob())->dailyAt('00:00');
```
Comment (Zelig880) [inline app/Console/Kernel.php:31]: I did not mind for this to be at 12.00. I really personally dislike when I receive email in the night! :)

File: app/Console/Kernel.php [hunk 3]
```diff
@@ -26,9 +26,9 @@ class Kernel extends ConsoleKernel
      */
     protected function schedule(Schedule $schedule)
     {
-        $schedule->job(new PickupBookJob())->dailyAt('12:00');
+        $schedule->job(new PickupBookJob())->hourly();
```
Comment (Zelig880) [inline app/Console/Kernel.php:29]: If we send this every hour, it will send 23 emails for PickupBookMail

**Review by Zelig880 — CHANGES_REQUESTED** [review]
Just a couple of comments.

**Comment by J-hon** [comment]
@Zelig880 
I'll make the necessary changes.

File: app/Console/Kernel.php (hunk 2 above)
Comment (J-hon) [reply app/Console/Kernel.php:31]: The reason I made it hourly was simply cos of the request being open for you 24 hours... So it has to be run every hour to capture that but I think I'll just create a separate cron job for that then.

Comment (J-hon) [reply app/Console/Kernel.php:31]: @Zelig880 
The reason I made it hourly was simply cos of the request being open for you 24 hours... So it has to be run every hour to capture that but I think I'll just create a separate cron job for that then.

Comment (J-hon) [reply app/Console/Kernel.php:31]: @Zelig880 
Ohh 
Sorry, I was talking about something else.

File: app/Console/Kernel.php (hunk 3 above)
Comment (J-hon) [reply app/Console/Kernel.php:29]: @Zelig880
The reason I made it hourly was simply cos of the request being open for you 24 hours... So it has to be run every hour to capture that but I think I'll just create a separate cron job for that then.

**Review by Zelig880 — APPROVED** [review]

**Pull request merged**